aten/src/ATen/native/native_functions.yaml

```diff
@@ -6211,6 +6211,26 @@
     CPU: _svd_helper_cpu
     CUDA: _svd_helper_cuda
 
+# swapaxes, alias for transpose
+- func: swapaxes(Tensor(a) self, int axis1, int axis2) -> Tensor(a)
+  use_c10_dispatcher: full
+  variants: function, method
+
+- func: swapaxes_(Tensor(a!) self, int axis1, int axis2) -> Tensor(a!)
+  use_c10_dispatcher: full
+  variants: method
+  device_guard: False
```

Comment:
If we look at the dispatch entries for transpose:

```
- func: transpose.int(Tensor(a) self, int dim0, int dim1) -> Tensor(a)
  use_c10_dispatcher: full
  variants: function, method
  device_guard: False
  dispatch:
    DefaultBackend: transpose

- func: transpose_(Tensor(a!) self, int dim0, int dim1) -> Tensor(a!)
  use_c10_dispatcher: full
  variants: method
  device_guard: False
  dispatch:
    DefaultBackend: transpose_
```

We see that `device_guard: False` appears on both the in place and out-of-place variants and that the dispatch is `DefaultBackend: transpose` or `DefaultBackend: transpose_`. 

Note also that transpose starts its dim indexing at zero, so swapaxes should have axis0 and axis1, and swapdims should have dim0 and dim1, too.

Reply:
@mruberry So should we remove `device_guard` from the signature? Sorry didn't understand comment.


> Note also that transpose starts its dim indexing at zero, so swapaxes should have axis0 and axis1, and swapdims should have dim0 and dim1, too.

Sure. The reason for this was because numpy names argument `axis1, axis2`, but makes sense to be more similar to transpose.

Reply:
@mruberry I think I understand now. `device_guard` should be on both signatues. Thanks!

Reply:
Yep, this looks correct.